packages/rln/src/message.ts

```diff
@@ -0,0 +1,70 @@
+import type {
+  IDecodedMessage,
+  IMessage,
+  IRateLimitProof
+} from "@waku/interfaces";
+import * as utils from "@waku/utils/bytes";
+
+import { RLNInstance } from "./rln.js";
+import { epochBytesToInt } from "./utils/index.js";
+
+export function toRLNSignal(contentTopic: string, msg: IMessage): Uint8Array {
+  const contentTopicBytes = utils.utf8ToBytes(contentTopic ?? "");
+  return new Uint8Array([...(msg.payload ?? []), ...contentTopicBytes]);
+}
+
+export class RlnMessage<T extends IDecodedMessage> implements IDecodedMessage {
+  public pubsubTopic = "";
+
+  constructor(
+    public rlnInstance: RLNInstance,
+    public msg: T,
+    public rateLimitProof: IRateLimitProof | undefined
+  ) {}
+
+  public verify(roots: Uint8Array[]): boolean | undefined {
+    return this.rateLimitProof
+      ? this.rlnInstance.zerokit.verifyWithRoots(
+          this.rateLimitProof,
+          toRLNSignal(this.msg.contentTopic, this.msg),
+          ...roots
+        ) // this.rlnInstance.verifyRLNProof once issue status-im/nwaku#1248 is fixed
+      : undefined;
+  }
+
+  public verifyNoRoot(): boolean | undefined {
+    return this.rateLimitProof
+      ? this.rlnInstance.zerokit.verifyWithNoRoot(
+          this.rateLimitProof,
+          toRLNSignal(this.msg.contentTopic, this.msg)
+        ) // this.rlnInstance.verifyRLNProof once issue status-im/nwaku#1248 is fixed
```

Comment:
https://github.com/waku-org/nwaku/issues/1248 is closed -- we should update this